clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -121,11 +127,13 @@ public HeartbeatRequestManager(
         this.heartbeatState = new HeartbeatState(subscriptions, membershipManager, rebalanceTimeoutMs);
         this.heartbeatRequestState = new HeartbeatRequestState(logContext, time, 0, retryBackoffMs,
             retryBackoffMaxMs, rebalanceTimeoutMs);
+        this.pollTimer = time.timer(rebalanceTimeoutMs);
     }
 
     // Visible for testing
     HeartbeatRequestManager(
         final LogContext logContext,
+        final Time time,
```

Comment:
Why not passing in directly the poll timer instead of the mock time that is only used to create the poll timer. By passing in the poll timer, you can also remove method `pollTimer()` since that is only needed for testing.